### ISSUE TITLE
Flagging non-buggy and legal code

### ISSUE THREAD

**Issue by bentcoder** [kkHAIKE/contextcheck#10]
Hi,

I don't understand why this code should be flagged. Any explanation or realistic reason would be appreciated for going deep to raise the flag.

Thanks

```go
// github.com/go-chi/chi/v5
// parameter is set as normal using `"/{id}"`

func find(w http.ResponseWriter, r *http.Request) {
   id := chi.URLParam(r, "id")
}
```

```bash
Function `URLParam->URLParam` should pass the context parameter (contextcheck)
	id := chi.URLParam(r, "id")
```

Using:

- golangci-lint has version 1.49.0

**Comment by kkHAIKE** [comment]
I don't get similar output.

source:
```gp
package chitst

import (
  "net/http"
  "github.com/go-chi/chi/v5"
)

func find(w http.ResponseWriter, r *http.Request) {
   id := chi.URLParam(r, "id")
   _ = id
}
```

can you share the complete code?

**Comment by wawan93** [comment]
I've got a similar issue: the linter tells me to pass context to the function without any parameters. 
```
Function `Close[staff-transactions/internal/queries/pg.ReviseReportPaymentsRow]->Close->close[T]->close` should pass the context parameter (contextcheck) 
```

```go
for rows.Next() {
	select {
	case <-ctx.Done():
		err = rows.Close() //nolint:contextcheck
		if err != nil {
			// ...
		}
		return
	default:
		// ...
	}
}

```

golangci-lint version v1.55.1

**Comment by kkHAIKE** [comment]
> ```go
> //nolint:contextcheck
> ```

You can add //nolint:contextcheck in the comment of the Close method you specified for this case, please refer to README.md.

**Comment by kkHAIKE** [comment]
> I've got a similar issue: the linter tells me to pass context to the function without any parameters.
> 
> ```
> Function `Close[staff-transactions/internal/queries/pg.ReviseReportPaymentsRow]->Close->close[T]->close` should pass the context parameter (contextcheck) 
> ```
> 
> ```go
> for rows.Next() {
> 	select {
> 	case <-ctx.Done():
> 		err = rows.Close() //nolint:contextcheck
> 		if err != nil {
> 			// ...
> 		}
> 		return
> 	default:
> 		// ...
> 	}
> }
> ```
> 
> golangci-lint version v1.55.1

You can add //nolint:contextcheck in the comment of the Close method you specified for this case, please refer to README.md.
